dev/functional/tests/cypress/integration/02-institution/03-institution-profile.spec.ts

```diff
@@ -0,0 +1,89 @@
+import DashboardInstitutionObject from "../../page-objects/Institution-objects/DashboardInstitutionObject";
+import InstitutionProfileObject from "../../page-objects/Institution-objects/InstitutionProfileObject";
+import InstitutionCustomCommand from "../../custom-command/institution/InstitutionCustomCommand";
+
+describe("Institution Profile", () => {
+  const dashboardInstitutionObject = new DashboardInstitutionObject();
+  const institutionObject = new InstitutionProfileObject();
+  const institutionCustomCommand = new InstitutionCustomCommand();
+
+  const url = Cypress.env("institutionURL");
+
+  beforeEach(() => {
+    cy.visit(url);
+    institutionCustomCommand.loginInstitution();
+  });
+
+  it("Verify that user redirect to institution profile page", () => {
+    dashboardInstitutionObject.dashboardButton().click();
+    dashboardInstitutionObject.manageInstitutionButton().click();
+    dashboardInstitutionObject.locationVerifyText().should("be.visible");
+    institutionObject.institutionDetailsButton().click();
+  });
+
+  it("Check that legal operating name text field should be disable or not", () => {
+    dashboardInstitutionObject.dashboardButton().click();
+    dashboardInstitutionObject.manageInstitutionButton().click();
+    dashboardInstitutionObject.locationVerifyText().should("be.visible");
+    institutionObject.institutionDetailsButton().click();
+    institutionObject.legalOperatingNameInputText().should("be.disabled");
+  });
+
+  it("Clicking on the save button without filling out the required fields", () => {
```

Comment:
The institution profile has different behaviors for different users at different moments, is all of this taken into consideration?

Reply:
Yes.